### ISSUE TITLE
Failing test 06/21/2017

### ISSUE THREAD

**Issue by kevinthesun** [dmlc/keras#65]
Require sparse support:

| Test cases
|---|
| test_sparse_mlp  |
| test_sparse_dot |
| test_sparse_concat |

Require rnn symbolic loop implementation:

| Test cases
|---|
| test_dynamic_behavior[SimpleRNN] |
| test_dynamic_behavior[GRU] |
| test_dynamic_behavior[LSTM] |
| imdb_lstm |
| conv_lstm |

Utility function issue, not major blocker:

| Test cases | Reason
|---|---|
| test_value_manipulation | Print tensor should return tensor |
| test_gradient | Require implementation of symbol gradient |
| test_function | Require updating variables |
| test_switch | Require implementation of switching between two operations |
| test_random_binomial | Require implementation of random binomial |
| test_ctc | Require implementation of ctc_batch_cost |
| test_map | Require implementation of mappinf function over elements |
| test_foldl | Require implementation of reducing elements using function |
| test_foldr | Require implementation of reducing elements using function |
| test_Eigenvalue_reg | Not supported |

Function incompatible with mxnet:

| Test cases | Reason
|---|---|
| test_nn_operations | MXNet Categorical_crossentropy doesn't support from_logits |
| test_arange | Keras requires that when start >= stop and step > 0, this function should return an empty sequence. Currently mxnet returns error |
| test_batchnorm_mode_0_or_2 | Currently keras mxnet doesn't support batchnorm mode 2 |
| test_shared_batchnorm | Currently keras mxnet doesn't support batchnorm mode 2 |
| variational_autoencoder_deconv | Too big target shape for mx.sym.deconvolution operator |
| mnist_acgan | For mxnet backend, parameters are not shared between concatenated model and other separate models. So user needs to directly train generator instead of combine generator and discriminator to one model. |
| test_sequential_model_saving | Optimizer states not preserved when saving/loading model |
| mnist_net2net | mxnet doesn't support set weights to larger shape |

Test case image ordering issue. Passing after modifying test cases. These are actually not failing tests:

| Test cases
|---|
| test_image_classification |
| test_conv2d |
| test_conv3d |
| test_atrous_conv_2d |
| test_averagepooling_2d |
| test_zero_padding_3d	|
| test_TimeDistributed |
